torch/nn/parallel/distributed.py

```diff
@@ -139,10 +138,16 @@ def __init__(self, module, device_ids=None, output_device=None, dim=0):
             if bucket_bytes >= bucket_bytes_cap:
                 self.bucket_sizes.append(0)
                 bucket_bytes = 0
-            self.bucket_sizes[-1] += 1
             for p in param_tuple:
-                self.bucket_map[p] = len(self.bucket_sizes) - 1
-            bucket_bytes += p.numel() * p.element_size()
+                add_to_bucket = False
+                if p.requires_grad:
+                    self.bucket_map[p] = len(self.bucket_sizes) - 1
+                    add_to_bucket = True
+            if add_to_bucket:
+                bucket_bytes += p.numel() * p.element_size()
+                self.bucket_sizes[-1] += 1
+        if self.bucket_sizes[-1]==0:
```

Comment:
If you modify the loop to skip the iteration at the very beginning this will become unnecessary as well.